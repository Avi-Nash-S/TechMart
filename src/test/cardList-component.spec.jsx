import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enzymeSerializer from "enzyme-to-json/serializer";
import CardListComponent from "../components/cardList-component";
import { products } from "../static/test-data/constant-test-data";
expect.addSnapshotSerializer(enzymeSerializer);

configure({ adapter: new Adapter() });

describe("<CardListComponent/>", () => {
  it("Should match the component snapshot", () => {
    const wrapper = shallow(
      <CardListComponent
        products={products}
        isLoading={false}
        history={[]}
        formView={false}
        onFormEdit={onFormEdit}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render in DOM with correct Inputs", () => {
    const wrapper = shallow(
      <CardListComponent
        products={products}
        isLoading={false}
        history={[]}
        formView={false}
        onFormEdit={onFormEdit}
      />
    );
    const container = wrapper.find(".lp-container");
    expect(container.length).toEqual(1);
    const cardComponents = wrapper.find("CardComponent");
    expect(cardComponents.length).toEqual(8);
    const cardComponentsPropCheck = wrapper.find("CardComponent").at(7);
    expect(cardComponentsPropCheck.props().history).toEqual([]);
  });
});

const onFormEdit = () => console.log("onFormEdit");
