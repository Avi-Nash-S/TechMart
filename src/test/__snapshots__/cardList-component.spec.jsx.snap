// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardListComponent/> Should match the component snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    className="lp-container"
  >
    <CardComponent
      history={Array []}
      isLoading={false}
      key="0"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": false,
          "price": "$49.00",
          "productId": "0ff4d723-9973-4450-8f3b-f98bfb8ebd10",
          "productImage": "/images/image7.jpeg",
          "productName": "SquareTrade 4-Year TV Protection Plan",
          "reviewCount": 0,
          "reviewRating": 0,
          "shortDescription": "<ul>
	<li>Protection against mechanical/electrical failures from normal use</li>
	<li>PLUS members receive an additional 12 months of coverage</li>
	<li>100% parts &amp; labor, no deductibles</li>
	<li>On-site service for TVs 37&rdquo; and above, free 2-way shipping for smaller TVs</li>
	<li>24/7 customer support</li>
</ul>
",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="1"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p><b>The only DVR that will sweep you off your couch. </b><br />
TiVo Roamio&trade; Pro gives you the power to watch any show, anywhere, anytime. Stream live or recorded shows directly to your iPad&reg; or iPhone&reg; around the house, or around the world. You can even download your shows and take them to go. Plus, you no longer need to switch inputs to access all the cool stuff happening on the web. The Roamio Pro serves up TV and streaming apps from a single menu, with one remote. Built with the one-of-a-kind features that make TiVo users passionately loyal, the powerful Roamio Pro records six shows at once and up to 450 hours of HD programming.</p>

<p style=\\"font-size: 12.727272033691406px;\\"><b>What you need to Get Started</b></p>

<ul style=\\"font-size: 12.727272033691406px;\\">
	<li>Subscription to the TiVo service via broadband connection</li>
	<li>Cable TV service, Verizon FiOS (Does not support satellite, AT&amp;T U-verse or antenna)</li>
	<li>One Multistream CableCARD decoder from your cable company</li>
</ul>

<p style=\\"font-size: 12.727272033691406px;\\"><br />
<b>Included in Box:</b></p>

<ul style=\\"font-size: 12.727272033691406px;\\">
	<li>TiVo Roamio Pro</li>
	<li>TiVo Roamio remote</li>
	<li>Quick-start guide</li>
	<li>Power cord</li>
</ul>

<p style=\\"font-size: 12.727272033691406px;\\"><br />
<b>Optional Accessories:</b></p>

<ul style=\\"font-size: 12.727272033691406px;\\">
	<li>TiVo Mini</li>
	<li>eSATA external drive</li>
</ul>
",
          "price": "$448.00",
          "productId": "0412c77e-204e-47f0-a7ff-27f4aba3416b",
          "productImage": "/images/image5.jpeg",
          "productName": "TiVo Roamio Pro",
          "reviewCount": 1,
          "reviewRating": 5,
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="2"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p>With 4K UHD resolution you&rsquo;ll enjoy a picture with 4X the detail of Full HD. Watch and play your way with the advanced Samsung Smart TV platform that lets you quickly and easily access your favorite content. Upgrade all lower resolution media to a stunning near ultra high-definition experience with enhanced detail and optimized picture quality. Experience UHD picture quality with deeper blacks, purer whites, brighter colors, and enhanced detail in every image. Enjoy improved fast-action moving picture resolution at Motion Rate 120 with outstanding refresh rate, processing speed and backlight technology. The screen mirroring feature allows you to mirror your phone or other compatible mobile device&rsquo;s screen onto the TVs screen wirelessly instead of your devices smaller screen for showing content, media playback, or other function. Have your Samsung Smart TV act as an alarm when synchronized with your other Samsung mobile devices. Use the large screen to display important items such as the time, weather, and your daily schedule.</p>
",
          "price": "$928.00",
          "productId": "0150f9b5-8918-4fd1-92b3-fc032cc6c684",
          "productImage": "/images/image5.jpeg",
          "productName": "Samsung 50\\" Class 4K Ultra HD LED Smart TV - UN50JU650DFXZA",
          "reviewCount": 2,
          "reviewRating": 4.5,
          "shortDescription": "<ul>
	<li>Resolution: 4K Ultra HD&nbsp;</li>
	<li>Refresh Rate: 60Hz w/ 120 Clear Motion Rate</li>
	<li>HDMI Inputs: 4</li>
</ul>
",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="3"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p>The Knightsbridge 58&nbsp;Inch TV&nbsp;Stand is a solid wood media center with rustic appeal. Designed to accommodate most wide screen TVs, this console has a solid wood center shelf for open access to electronic components. Four generous storage drawers keep movie collections and gaming accessories in one convenient place. The unfinished salvaged teak planks have all the original marks and imperfections, giving this TV console its unique weathered appearance that will only improve with daily use and age.</p>
",
          "price": "$1199.00",
          "productId": "0e56dd23-29a0-4d0e-882a-6f7d851d7ec4",
          "productImage": "/images/image8.jpeg",
          "productName": "Knightsbridge Reclaimed Wood 58 Inch TV Stand",
          "reviewCount": 0,
          "reviewRating": 0,
          "shortDescription": "<ul style=\\"font-size: 13px; line-height: 20.7999992370605px;\\">
	<li>HFSC approved Pine framework</li>
	<li>Natural reclaimed teak planks</li>
	<li>Made from solid wood boards, no veneers or particle boards used</li>
</ul>
",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="4"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p>The AQUOS LC-43LE653U gives you gorgeous, reliable Smart TV performance with technologies like AquoMotion&trade; 120 for smooth motion and reduced blur and SmartCentral&trade; Apps for smart TV access. Add to that its ultra-narrow bezel and sleek modern design, Wallpaper mode, three HDMI inputs including MHL, and built-in 20w audio, and you&rsquo;ve got a great all-around Full HD LED Smart TV.</p>
",
          "price": "$368.00",
          "productId": "04721d49-835e-40d7-aabb-03f9c02fe173",
          "productImage": "/images/image8.jpeg",
          "productName": "Sharp 43\\" Class 1080p LED Smart HDTV - LC-43LE653U",
          "reviewCount": 1,
          "reviewRating": 1,
          "shortDescription": "<ul>
	<li>Resolution: 1080p&nbsp;</li>
	<li>Refresh Rate: 60Hz w/ AquoMotion 120</li>
	<li>HDMI Inputs: 3</li>
</ul>
",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="5"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "Whether youre a casual video game player, or serious power gamer, Atlantics Spyder TV Gaming Stand is the perfect companion to all your video gaming needs. It's packed with tons of cool features to keep your gaming equipment organized and provides a versatile flat screen TV stand.",
          "price": "$119.88",
          "productId": "28e689d1-31f1-45d4-8684-16e0e1c80997",
          "productImage": "/images/image7.jpeg",
          "productName": "Spyder TV Gaming Stand for up to 42\\" TV - Black",
          "reviewCount": 0,
          "reviewRating": 0,
          "shortDescription": "The ideal flat screen TV stand and gaming console/accessory storage solution. ",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="6"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p>Introducing the all-new 2014 E-Series 23&rdquo; (22.95&rdquo; diag.) Razor LED TV with an ultra-narrow frame and vibrant LED-lit picture. Edge-lit Razor LED backlighting delivers brilliant picture quality in an ultra-thin design. Enjoy high definition TV in crisp, clear resolution. With a near borderless design, a thinner side profile and space-saving, slimmer base, the new E-Series 23&rdquo; Full-Array LED TV is a perfect upgrade to any room.</p>

<p>VIZIO E-Series: Picture-Perfect Brilliance.</p>
",
          "price": "$134.88",
          "productId": "1f80e11b-1412-4990-813a-f068e1122873",
          "productImage": "/images/image9.jpeg",
          "productName": "VIZIO 23\\" Class 720p LED HDTV - E231-B1",
          "reviewCount": 2,
          "reviewRating": 3,
          "shortDescription": "<ul>
	<li>Resolution: 720p</li>
	<li>Refresh Rate: 60Hz</li>
	<li>HDMI Input: 1</li>
</ul>
",
        }
      }
    />
    <CardComponent
      history={Array []}
      isLoading={false}
      key="7"
      onFormEdit={[Function]}
      product={
        Object {
          "inStock": true,
          "longDescription": "<p><b>LED</b><br />
LED uses tiny, energy-efficient light-emitting diodes to illuminate the picture. The new standard in HDTVs, LED is superior to older CCFL technology, making slimmer televisions possible, with amazing brightness, clarity and color detail.</p>

<p><span style=\\"line-height: 1.6em;\\"><b>Triple XD Engine</b><br />
LG&rsquo;s exceptional picture quality is further enhanced by the Triple XD Engine, which processes images with even greater precision. The latest Triple XD Engine enables more natural color expression, deeper contrast and more refined motion so that viewers can enjoy a more lifelike picture.</span></p>

<p><span style=\\"line-height: 1.6em;\\"><b>Smart Energy Saving<br />
</b></span><span style=\\"line-height: 1.6em;\\">This valuable feature includes a backlight control to adjust the brightness, a screen-off function that turns the screen off to play audio only, and the Standby Mode Zero function that allows the TV to effectively hibernate, using zero electricity.</span></p>
",
          "price": "$217.88",
          "productId": "218ca04a-0a9f-42b3-9fa5-1ca592fc7d29",
          "productImage": "/images/image5.jpeg",
          "productName": "LG 32\\" Class 720P LED HDTV - 32LF500B",
          "reviewCount": 16,
          "reviewRating": 4.8125,
          "shortDescription": "<ul>
	<li>Resolution: 720p</li>
	<li>Refresh Rate: 60Hz</li>
	<li>HDMI Inputs: 3</li>
</ul>
",
        }
      }
    />
  </div>
</div>
`;
